.github/workflows/integration-testing.yml

```diff
@@ -0,0 +1,113 @@
+name: Integration Testing
+on: [push, pull_request]
+
+jobs:
+  build_Java_Agent:
+    name: Build AWS OTel Java agent
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - uses: gradle/wrapper-validation-action@v1
+
+      - name: Build with Gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          arguments: build --stacktrace -PenableCoverage=true
+
+      - name: Upload agent artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: aws-opentelemetry-agent.jar
+          path: otelagent/build/libs/aws-opentelemetry-agent-*.jar
+
+  build_Web_App:
+    name: Build Spring Web Application
+    needs: build_Java_Agent
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Pull in source code from WebApp Github repo
+        uses: actions/checkout@v2
+        with:
+          repository: aws-observability/aws-otel-test-framework-sample-apps
+          token: ${{  secrets.PAT }}
```

Comment:
You'll probably have to get this `PAT` on the base repo (not your fork) if you want this to work because of some Github Actions [security policies](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories) which sends the workflow to the base repo

Reply:
Yes but I am trying to avoid having my PAT in org wide repo. Now, we are adding sample apps to SDK repo so we don't have to import this repo.